tests/golem/task/test_taskcomputeradapter.py

```diff
@@ -0,0 +1,371 @@
+from pathlib import Path
+from unittest import mock
+
+from golem_messages.message import ComputeTaskDef
+from twisted.internet import defer
+from twisted.trial.unittest import TestCase as TwistedTestCase
+
+from golem.task.envmanager import EnvironmentManager
+from golem.task.taskcomputer import TaskComputerAdapter
+from golem.task.taskserver import TaskServer
+from tests.factories.taskserver import ClientConfigDescriptor
+
+
+class TaskComputerAdapterTestBase(TwistedTestCase):
+
+    @mock.patch('golem.task.taskcomputer.IntStatsKeeper')
+    @mock.patch('golem.task.taskcomputer.TaskComputer')
+    @mock.patch('golem.task.taskcomputer.NewTaskComputer')
+    def setUp(self, new_task_computer, old_task_computer, int_stats_keeper):
+        self.new_computer = new_task_computer()
+        self.old_computer = old_task_computer()
+        self.int_stats_keeper = int_stats_keeper()
+        config_desc = ClientConfigDescriptor()
+        config_desc.accept_tasks = True
+        config_desc.in_shutdown = False
+        self.task_server = mock.Mock(
+            spec=TaskServer,
+            config_desc=config_desc,
+            task_keeper=mock.Mock()
+        )
+        self.env_manager = mock.Mock(spec_set=EnvironmentManager)
+        self.adapter = TaskComputerAdapter(
+            task_server=self.task_server,
+            env_manager=self.env_manager
+        )
+
+
+class TestInit(TaskComputerAdapterTestBase):
+
+    def test(self):
```

Comment:
Not sure if this is required by pytest but i think its neat to have a name for the test
```suggestion
    def test_init(self):
```